File: java-sample/src/main/java/com/moengage/sample/java/inbox/customadapter/CustomInboxAdapter.java
package com.moengage.sample.java.inbox.customadapter;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import com.moengage.inbox.core.model.InboxMessage;
import com.moengage.inbox.ui.adapter.InboxAdapter;
import com.moengage.inbox.ui.adapter.InboxListAdapter;
import com.moengage.inbox.ui.adapter.ViewHolder;
import com.moengage.sample.java.databinding.CustomInboxItemViewBinding;
import org.jetbrains.annotations.NotNull;

/**
 * To use your own custom InboxAdapter, extend [InboxAdapter]and bind
 * your customised [ViewHolder] class.
 */
public class CustomInboxAdapter extends InboxAdapter {

  @Override public long getItemId(int i) {
    return i;
  }

  @Override public int getItemViewType(int i, @NotNull InboxMessage inboxMessage) {
    return 0;
  }

  @Override public void onBindViewHolder(@NotNull ViewHolder viewHolder, int i,
      @NotNull InboxMessage inboxMessage, @NotNull InboxListAdapter inboxListAdapter) {
    ((CustomInboxViewHolder) viewHolder).onBind(i, inboxMessage, inboxListAdapter);
  }

  @NotNull @Override public ViewHolder onCreateViewHolder(@NotNull ViewGroup viewGroup, int i) {
    return new CustomInboxViewHolder(
        CustomInboxItemViewBinding.inflate(LayoutInflater.from(viewGroup.getContext()), viewGroup,
            false));
  }
}

File: kotlin-sample/src/main/java/com/moengage/sample/kotlin/inbox/customadapter/CustomInboxAdapter.kt
package com.moengage.sample.kotlin.inbox.customadapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.moengage.inbox.core.model.InboxMessage
import com.moengage.inbox.ui.adapter.InboxAdapter
import com.moengage.inbox.ui.adapter.ViewHolder
import com.moengage.inbox.ui.adapter.InboxListAdapter
import com.moengage.sample.kotlin.databinding.CustomInboxItemViewBinding

/**
 * To use your own custom InboxAdapter, extend [InboxAdapter]and bind
 * your customised [ViewHolder] class.
 *
 */
class CustomInboxAdapter : InboxAdapter() {

    /**
     * @return custom [ViewHolder] object - [CustomInboxViewHolder]
     */
    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
        return CustomInboxViewHolder(
            CustomInboxItemViewBinding.inflate(
                LayoutInflater.from(viewGroup.context),
                viewGroup,
                false
            )
        )
    }

    override fun getItemViewType(position: Int, inboxMessage: InboxMessage): Int {
        return 1
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun onBindViewHolder(
        viewHolder: ViewHolder,
        position: Int,
        inboxMessage: InboxMessage,
        inboxListAdapter: InboxListAdapter
    ) {
        (viewHolder as CustomInboxViewHolder).onBind(position, inboxMessage, inboxListAdapter)
    }
}